### PR TITLE
CORTX-31464 - Fix cortx-re codacy issues

### PR DESCRIPTION
**Pull request by subhalaxmisa**
Signed-off-by: subhalaxmisa <subhalaxmi.sahoo@seagate.com>

# Problem Statement
- Fix Cortx-re Codacy issues by reviewing the suggested changes in codeacy : https://app.codacy.com/gh/Seagate/cortx-re/dashboard

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [x] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: solutions/community-deploy/CORTX-Deployment.md [hunk 1]
```diff
@@ -109,4 +109,4 @@ Run IO Sanity on your CORTX Cluster to validate if you are able to create bucket
 
 Tested by:
 
-* May 06, 2022: Rahul Shenoy (rahul.shenoy@seagate.com) - Windows + VMware Workstation 16 + CentOS 7.9 Linux
\ No newline at end of file
+* May 06, 2022: Rahul Shenoy (rahul.shenoy@seagate.com) - Windows '+' VMware Workstation 16 '+' CentOS 7.9 Linux
```
Comment (gauravchaudhari02) [inline solutions/community-deploy/CORTX-Deployment.md:112]: why these single quotes added for `+` sign? please remove it

**Comment by gauravchaudhari02** [comment]
@subhalaxmisa, please write problem statement in detail and mention how you resolved it in design section. refer other PR's for details

File: solutions/community-deploy/CORTX-Deployment.md (hunk 1 above)
Comment (subhalaxmisa) [reply solutions/community-deploy/CORTX-Deployment.md:112]: @gauravchaudhari02 Please find the codeacy error : https://app.codacy.com/gh/Seagate/cortx-re/file/71107647120/issues/source?bid=27005559&fileBranchId=27005559#l112

Comment (gauravchaudhari02) [reply solutions/community-deploy/CORTX-Deployment.md:112]: what if we put `,` instead of `+` sign?

Comment (subhalaxmisa) [reply solutions/community-deploy/CORTX-Deployment.md:112]: We can. Let me try that as well.

File: solutions/kubernetes/ceph-binary-build.sh [hunk 2]
```diff
@@ -61,7 +61,7 @@ function prereq() {
 
     pushd "$BUILD_LOCATION"
         add_common_separator "Removing previous files"
-        rm -rvf *
+        rm -rvf ./*
```
Comment (subhalaxmisa) [inline solutions/kubernetes/ceph-binary-build.sh:64]: Let me know if we are good to make this kind of changes suggested by codacy. Link : https://app.codacy.com/gh/Seagate/cortx-re/file/71107647137/issues/source?bid=27005559&fileBranchId=27005559#l64

File: solutions/community-deploy/CORTX-Deployment.md (hunk 1 above)
Comment (subhalaxmisa) [reply solutions/community-deploy/CORTX-Deployment.md]: So we are good with the above changes. Hence resolving the conversation.

File: docker/cortx-deploy/Dockerfile [hunk 3]
```diff
@@ -61,18 +61,18 @@ RUN yum install yum-utils -y \
...
     && yum clean all && rm -rf /var/cache/yum
 
-RUN if [[ "$CORTX_IMAGE" = "cortx-rgw" ]] || [[ "$CORTX_IMAGE" = "cortx-all" ]] || [[ "$CORTX_IMAGE" = "cortx-data" ]]; then yum --nogpgcheck -y --disablerepo="EOS_Rocky_8_OS_x86_64_Rocky_8" install libfabric-1.11.2; fi \
+RUN if [[ "$(CORTX_IMAGE)" = "cortx-rgw" ]] || [[ "$(CORTX_IMAGE)" = "cortx-all" ]] || [[ "$(CORTX_IMAGE)" = "cortx-data" ]]; then yum --nogpgcheck -y --disablerepo="EOS_Rocky_8_OS_x86_64_Rocky_8" install libfabric-1.11.2; fi \
     && yum install --nogpgcheck -y  $(while read -r component; do grep "$component" < /opt/seagate/cortx/third-party-rpms.txt; done </opt/seagate/cortx/cortx-component-rpms.txt | sed 's/#.*//g') \
     && yum clean all && rm -rf /var/cache/yum
     
 RUN yum install --nogpgcheck -y  $(sed 's/#.*//g' /opt/seagate/cortx/cortx-component-rpms.txt) \
-    && if [[ "$CORTX_IMAGE" = "cortx-rgw" ]]; then yum install --nogpgcheck -y  $(sed 's/#.*//g' /opt/seagate/cortx/ceph-component-rpms.txt); fi \
+    && if [[ "$(CORTX_IMAGE)" = "cortx-rgw" ]]; then yum install --nogpgcheck -y  $(sed 's/#.*//g' /opt/seagate/cortx/ceph-component-rpms.txt); fi \
```
Comment (shailesh-vaidya) [inline docker/cortx-deploy/Dockerfile:75]: Please validate CORTX container images generation and deployment with these changes.

File: jenkins/internal-ci/centos-7.8.2003/main/cortx-ha.groovy [hunk 4]
```diff
@@ -218,11 +218,11 @@ def getAuthor(issue) {
     def author= ""
     def response = ""
     // Grab build information
-    for (int i = 0; i < changeLogSets.size(); i++){
+    for (int i = 0; i < changeLogSets.size(); i++) {
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/centos-7.8.2003/main/cortx-ha.groovy:221]: Please test one job with this change to ensure nothing breaks.

File: jenkins/internal-ci/centos-7.9.2009/branch/motr.groovy [hunk 5]
```diff
@@ -151,7 +151,7 @@ pipeline {
                                     string(name: 'branch', value: "${branch}")
                                 ]
                                 env.HARE_BUILD_NUMBER = hareBuild.number
-                            }catch (err){
+                            }catch (err) {
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/centos-7.9.2009/branch/motr.groovy:154]: Should we also change `}catch` to `} catch`

File: solutions/community-deploy/cloud/AWS/setup.sh [hunk 6]
```diff
@@ -30,7 +30,7 @@ for nic in eth1 eth2
 do
    MAC=$(ifconfig $nic | grep -o -E '([[:xdigit:]]{2}:){5}[[:xdigit:]]{2}')
    cp /etc/sysconfig/network-scripts/ifcfg-eth0 /etc/sysconfig/network-scripts/ifcfg-$nic
-   sed -i 's/eth0/$nic/g' /etc/sysconfig/network-scripts/ifcfg-$nic
+   sed -i "s/eth0/$nic/g" /etc/sysconfig/network-scripts/ifcfg-$nic
```
Comment (shailesh-vaidya) [inline solutions/community-deploy/cloud/AWS/setup.sh:33]: Please check that scripts works with this change

File: solutions/kubernetes/ceph-binary-build.sh (hunk 2 above)
Comment (shailesh-vaidya) [reply solutions/kubernetes/ceph-binary-build.sh:64]: Please check that scripts works with this change

File: solutions/kubernetes/ceph-binary-build.sh [hunk 7]
```diff
@@ -144,7 +144,7 @@ function ceph_build() {
                 popd
 
                 add_common_separator "List generated binary packages (*.deb)"
-                ls *.deb
+                ls ./*.deb
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/ceph-binary-build.sh:147]: Please check that scripts works with this change

File: solutions/kubernetes/ceph-binary-build.sh [hunk 8]
```diff
@@ -176,7 +176,7 @@ function ceph_build() {
 
                     pushd rpmbuild
                         add_common_separator "Start Build"
-                        rpmbuild --define "_topdir `pwd`" -ba SPECS/ceph.spec
+                        rpmbuild --define "_topdir $(pwd)" -ba SPECS/ceph.spec
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/ceph-binary-build.sh:179]: Please check that the scripts work with this change

File: solutions/kubernetes/ceph-deploy-functions.sh [hunk 9]
```diff
@@ -103,7 +103,7 @@ function install_ceph() {
         rocky)
             pushd /root/RPMS # subject to change until binaries are fetched from a central repo
                 mv noarch/*.rpm . && mv x86_64/*.rpm . && rmdir noarch/ x86_64/
-                rpm -ivh *.rpm
+                rpm -ivh ./*.rpm
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/ceph-deploy-functions.sh:106]: Please check that the scripts work with this change

File: solutions/kubernetes/cluster-functions.sh [hunk 10]
```diff
@@ -219,7 +219,7 @@ function install_prerequisites() {
             CALICO_IMAGE_VERSION=$(grep 'docker.io/calico/cni' calico-$CALICO_PLUGIN_VERSION.yaml | uniq | awk -F':' '{ print $3}')	
             wget -c https://github.com/projectcalico/calico/releases/download/$CALICO_IMAGE_VERSION/release-$CALICO_IMAGE_VERSION.tgz -O - | tar -xz || throw $Exception
             pushd release-$CALICO_IMAGE_VERSION/images 
-                ls *tar | xargs -I{} docker image load -i {} || throw $Exception
+                ls ./*tar | xargs -I{} docker image load -i {} || throw $Exception
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/cluster-functions.sh:222]: Please check that the scripts work with this change

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: docker/cortx-deploy/Dockerfile [hunk 11]
```diff
@@ -61,18 +61,18 @@ RUN yum install yum-utils -y \
 COPY ./*.txt ./${CORTX_IMAGE}/*.txt ./compatibility.info ./generate-release-info.sh /opt/seagate/cortx/
 
 RUN yum install --nogpgcheck -y python3-pip \
-    && pip3 install yq \
+    && pip3 install yq==1.6 \
```
[the remark this answers is not in the document]
Comment (subhalaxmisa) [reply docker/cortx-deploy/Dockerfile:64]: Yes there was recommendation to use the version, so I picked up this version from my local. Let me know if you want me to change it.

File: jenkins/internal-ci/centos-7.8.2003/main/cortx-ha.groovy (hunk 4 above)
Comment (subhalaxmisa) [reply jenkins/internal-ci/centos-7.8.2003/main/cortx-ha.groovy:221]: According to codacy this the best practice.

File: jenkins/internal-ci/centos-7.9.2009/branch/motr.groovy (hunk 5 above)
Comment (subhalaxmisa) [reply jenkins/internal-ci/centos-7.9.2009/branch/motr.groovy:154]: This change` } catch` was not suggested.

File: jenkins/internal-ci/centos-7.8.2003/main/cortx-ha.groovy (hunk 4 above)
Comment (shailesh-vaidya) [reply jenkins/internal-ci/centos-7.8.2003/main/cortx-ha.groovy:221]: Yes, but the code should also with Codacy's recommended changes. We earlier observed that sometime Codacy recommendations caused script failures.

File: docker/cortx-deploy/Dockerfile (hunk 11 above)
[the remark this answers is not in the document]
Comment (shailesh-vaidya) [reply docker/cortx-deploy/Dockerfile:64]: Can you please point me to the doc/Jira ticket for yq version recommendation

[the remark this answers is not in the document]
Comment (subhalaxmisa) [reply docker/cortx-deploy/Dockerfile:64]: Any version you can any keep, it was asked to keep any version. So I added the version that I used to build cortx-rgw. Let me know if you want to change the version

File: jenkins/internal-ci/centos-7.8.2003/main/cortx-ha.groovy (hunk 4 above)
Comment (subhalaxmisa) [reply jenkins/internal-ci/centos-7.8.2003/main/cortx-ha.groovy:221]: We can test but executing each script will take a lot of time. Some scripts are dependent on some other scripts as well.

File: solutions/community-deploy/cloud/AWS/setup.sh (hunk 6 above)
Comment (subhalaxmisa) [reply solutions/community-deploy/cloud/AWS/setup.sh:33]: Yes this is working.

File: solutions/kubernetes/ceph-binary-build.sh (hunk 7 above)
Comment (subhalaxmisa) [reply solutions/kubernetes/ceph-binary-build.sh]: Removed this file

File: solutions/kubernetes/ceph-deploy-functions.sh [hunk 12]
```diff
@@ -89,7 +89,7 @@ function install_prereq() {
         ;;
         ubuntu)
             pushd /root/RPMS # subject to change until binaries are fetched from a central repo
-                dpkg -i *.deb     # this command will throw errors which is expected as it collects required dependencies for the installation
+                dpkg -i ./*.deb     # this command will throw errors which is expected as it collects required dependencies for the installation
```
[the remark this answers is not in the document]
Comment (subhalaxmisa) [reply solutions/kubernetes/ceph-deploy-functions.sh]: Removed this file

File: solutions/kubernetes/cluster-functions.sh (hunk 10 above)
Comment (subhalaxmisa) [reply solutions/kubernetes/cluster-functions.sh]: Removed this file

File: solutions/kubernetes/ceph-deploy-functions.sh (hunk 9 above)
Comment (subhalaxmisa) [reply solutions/kubernetes/ceph-deploy-functions.sh]: Remove this file

Comment (subhalaxmisa) [reply solutions/kubernetes/ceph-deploy-functions.sh]: Remove this file

File: docker/cortx-deploy/Dockerfile (hunk 11 above)
[the remark this answers is not in the document]
Comment (shailesh-vaidya) [reply docker/cortx-deploy/Dockerfile:64]: Please use `yq==2.14.0` . This is the latest release and it's getting installed by default. Refer https://pypi.org/project/yq/#history

[the remark this answers is not in the document]
Comment (subhalaxmisa) [reply docker/cortx-deploy/Dockerfile]: Done

[the remark this answers is not in the document]
Comment (subhalaxmisa) [reply docker/cortx-deploy/Dockerfile]: Tested the above docker file and it is working as expected. Please find the jenkins job: https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/Subha_WS/job/Docker_image_creation/6/consoleFull

File: solutions/kubernetes/ceph-binary-build.sh (hunk 8 above)
Comment (subhalaxmisa) [reply solutions/kubernetes/ceph-binary-build.sh]: Reverted it back

File: docker/cortx-deploy/Dockerfile [hunk 13]
```diff
@@ -61,18 +61,18 @@ RUN yum install yum-utils -y \
...
-    && pip3 install yq \
+    && pip3 install yq==2.14.0 \
     && while read component; do grep "$component" < /opt/seagate/cortx/python_requirements.txt; done </opt/seagate/cortx/cortx-component-rpms.txt | sed 's/#.*//g' > /opt/seagate/cortx/requirements.txt \
     && while read component; do grep "$component" < /opt/seagate/cortx/python_requirements.ext.txt; done </opt/seagate/cortx/cortx-component-rpms.txt | sed 's/#.*//g' > /opt/seagate/cortx/requirements.ext.txt \
     && pip3 install  --no-cache-dir --trusted-host $(echo $BUILD_URL | awk -F '/' '{print $3}') -i $BUILD_URL/python_deps -r /opt/seagate/cortx/requirements.txt -r /opt/seagate/cortx/requirements.ext.txt \
     && yum clean all && rm -rf /var/cache/yum
 
-RUN if [[ "$CORTX_IMAGE" = "cortx-rgw" ]] || [[ "$CORTX_IMAGE" = "cortx-all" ]] || [[ "$CORTX_IMAGE" = "cortx-data" ]]; then yum --nogpgcheck -y --disablerepo="EOS_Rocky_8_OS_x86_64_Rocky_8" install libfabric-1.11.2; fi \
+RUN if [[ "${CORTX_IMAGE}" = "cortx-rgw" ]] || [[ "${CORTX_IMAGE}" = "cortx-all" ]] || [[ "${CORTX_IMAGE}" = "cortx-data" ]]; then yum --nogpgcheck -y --disablerepo="EOS_Rocky_8_OS_x86_64_Rocky_8" install libfabric-1.11.2; fi \
```
Comment (shailesh-vaidya) [inline docker/cortx-deploy/Dockerfile:70]: Please create CORTX images using the custom-ci pipeline with your repo and branch as input - Refer - https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/
Test CORTX deployment with images generated from the above job.

File: solutions/kubernetes/ceph-binary-build.sh [hunk 14]
```diff
@@ -215,7 +215,7 @@ function ceph_build() {
...
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/ceph-binary-build.sh:218]: Please validate that script works after this change.

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
Please add Jenkins jobs used for validating changes on PR template.

File: solutions/kubernetes/ceph-binary-build.sh (hunk 14 above)
Comment (subhalaxmisa) [reply solutions/kubernetes/ceph-binary-build.sh]: As discussed reverted the changes.

File: docker/cortx-deploy/Dockerfile (hunk 13 above)
Comment (subhalaxmisa) [reply docker/cortx-deploy/Dockerfile:70]: 1. Created custom images using jenkins job : https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/
2. After images were build took the latest image and deployed cortx cluster : https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/

The Cortx deployment is failing for ha pods.

Comment (shailesh-vaidya) [reply docker/cortx-deploy/Dockerfile:70]: Thanks. I will validate this part. Please freeze any code changes to this PR now.

Comment (subhalaxmisa) [reply docker/cortx-deploy/Dockerfile:70]: Sure will do that.

Comment (shailesh-vaidya) [reply docker/cortx-deploy/Dockerfile:70]: Deployment is working. Validated with the below steps - 

- Image Generation - https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6323/console 

```
00:38:02.571  cortx-docker.colo.seagate.com/seagate/cortx-all:2.0.0-6323-custom-ci
00:38:02.571  cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-6323-custom-ci
00:38:02.571  cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-6323-custom-ci
00:38:02.571  cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-6323-custom-ci
```

- Deployment -  https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/5422/parameters/

image

File: docker/cortx-deploy/Dockerfile [hunk 15]
```diff
@@ -61,18 +61,18 @@ RUN yum install yum-utils -y \
 COPY ./*.txt ./${CORTX_IMAGE}/*.txt ./compatibility.info ./generate-release-info.sh /opt/seagate/cortx/
 
 RUN yum install --nogpgcheck -y python3-pip \
-    && pip3 install yq \
+    && pip3 install yq==2.14.0 \
```
Comment (gauravchaudhari02) [inline docker/cortx-deploy/Dockerfile:64]: @shailesh-vaidya, should we use below yq version here as well?
https://github.com/Seagate/cortx-re/blob/3bf8fdcf0eda4cc2a3829d6b918013b89cd0220c/solutions/kubernetes/cortx-deploy-functions.sh#L27
CC - @subhalaxmisa

Comment (shailesh-vaidya) [reply docker/cortx-deploy/Dockerfile:64]: @gauravchaudhari02 - YQ_VERSION=v4.25.1  is not available through pip. And anyways we are planning to remove `yq `from an image. So let this as it is.

**Review by gauravchaudhari02 — APPROVED** [review]

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**